### DISCUSSION TITLE
Social Media Tags

### DISCUSSION THREAD

**Discussion by gijoe2cool** [YOURLS/YOURLS#3635] [Q&A]
Ok, I've looked around and haven't found anything. It's also possible I'm looking for the wrong terms, but I digress. Here is my small story for anyone who can help me out.

I am coming over form Firebase, as they are getting rid of their dynamic url feature. Which irritates me, but I'll survive, I hope. While YOURLS at least gets me back in the game to making my own short urls with my own domain, it is lacking a key feature I came to rely on, social media stuff... And when I say stuff, I mean the meta data stuff that goes along with a link. Like when I post a link on social media it's not just a link the "link preview" pops up, and I can see the title I gave it, a picture, and a link description. 

So far as I've been digging, I haven't found a way to do that with YOURLS. Is this possible, and I just haven't seen a plugin to do it, or am I looking for the wrong thing?

Also while I have you all here, why can't I use capital links in my short urls? Or again, is this something I'm doing wrong on my end?

I have at least found a plugin that answers my question of bulk uploads, however that being said, if I want other things tagged on to the links for social media purposes then I'm sure the plugin I found for that wouldn't work anyway.

Anyway, thank you all for sitting through this, any help you have would be greatly appreciated.

**Comment by dgw** [comment]
I don't believe social media previews (Open Graph metadata) are technically possible given how YOURLS works. Your short links are redirected before the client ever sees any HTML from the YOURLS instance. See #2203.

> Also while I have you all here, why can't I use capital links in my short urls? Or again, is this something I'm doing wrong on my end?

https://yourls.org/docs/guide/essentials/configuration#yourls_url_convert

**Reply by dgw** [reply]
Looking at [the Firebase documentation](https://firebase.google.com/docs/dynamic-links/link-previews), I'd guess that it's doing some user-agent sniffing to send the metadata to "supported services" only and send a redirect to everyone else.

That should be possible with a plugin, but I don't know of any currently published that does so.

**Reply by gijoe2cool** [reply]
Thanks for the answer to the capitals! But I've gotten really used to using the firebase features, and it's spoiled me a bit when making my short links. 

I also haven't found any other url shortener that has the ability to do this, paid or otherwise. It almost seems it was a firebase only thing, or else I'm just missing something. I guess I may just have to get used to not having that feature anymore, but it was nice when posting to facebook/twitter(X).
